packages/teuchos/CMakeLists.txt

```diff
@@ -79,44 +87,58 @@ ENDIF()
 
 ASSERT_DEFINED(${PROJECT_NAME}_ENABLE_${PACKAGE_NAME}Numerics)
 IF (${PROJECT_NAME}_ENABLE_${PACKAGE_NAME}Numerics)
-
-  INCLUDE(CheckBlasFloat)
-  CHECK_BLAS_FLOAT(HAVE_TEUCHOS_BLASFLOAT)
-  IF(NOT HAVE_TEUCHOS_BLASFLOAT)
-    # try apple veclib first
-    IF(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
-      INCLUDE(CheckBlasFloatAppleVeclibBugfix)
-      CHECK_BLAS_FLOAT_APPLE_VECLIB_BUGFIX(HAVE_TEUCHOS_BLASFLOAT_APPLE_VECLIB_BUGFIX)
-      IF(HAVE_TEUCHOS_BLASFLOAT_APPLE_VECLIB_BUGFIX)
-        GLOBAL_SET(HAVE_TEUCHOS_BLASFLOAT 1)
-      ENDIF()
-    ENDIF()
-    # if veclib failed, general purpose solution
+  #
+  # The check needs to run code, so give cross-compiling users a bypass.
+  #
+  IF (DEFINED HAVE_TEUCHOS_BLASFLOAT)
+    MESSAGE(STATUS "You set HAVE_TEUCHOS_BLASFLOAT=${HAVE_TEUCHOS_BLASFLOAT}; this suggests you are cross compiling.  If you're not cross compiling, best practice is not to set this; Trilinos will figure it out for you.")
+  ELSE()
+    INCLUDE(CheckBlasFloat)
+    CHECK_BLAS_FLOAT(HAVE_TEUCHOS_BLASFLOAT)
     IF(NOT HAVE_TEUCHOS_BLASFLOAT)
-      INCLUDE(CheckBlasFloatReturnsDouble)
-      CHECK_BLAS_FLOAT_DOUBLE_RETURN(HAVE_TEUCHOS_BLASFLOAT_DOUBLE_RETURN)
```

Comment:
Following up on @bartlettroscoe's comment, I can't find this in the cmake [documentation](https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/VariablesListsStrings).

Reply:
cross compiling with cmake is typically handled with toolchain files. Not sure if that works in this case though. 

https://cmake.org/cmake/help/v3.6/manual/cmake-toolchains.7.html

Reply:
@jhux2, the function `CHECK_BLAS_FLOAT_DOUBLE_RETURN()` is our own custom function but it calls the standard CMake function `CHECK_CXX_SOURCE_RUNS()` which works with cache vars as described above.   That is how this is supposed to work.  You define your own configure-time checks that calls `CHECK_CXX_SOURCE_RUNS()` and then you can provide the value of the cmake cache var (shown as `HAVE_TEUCHOS_BLASFLOAT_DOUBLE_RETURN` in this case).  Make sense?

Reply:
@jhux2 Those custom checks live in `Trilinos/packages/teuchos/cmake`.

Reply:
The problem with `SOURCE_RUNS` is that it won't work with cross compiling.  If this can be coerced into a `try_compile` test, then you skirt the problem of compiling for a different architecture than the build system is running on.